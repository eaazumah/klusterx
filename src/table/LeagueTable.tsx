import { Badge, Container, Space, Table, Text } from '@mantine/core'
import React from 'react'
import styled from 'styled-components'
import data from '../data/data'
import useGenerateLeagueTable from './hooks/useGenerateLeagueTable'

const LeagueTable: React.FC = () => {
  const tableData = useGenerateLeagueTable(data)
  return (
    <Styled size='lg'>
      <Badge variant='dot'>League Table</Badge>
      <Space h='xl' />
      <Table striped highlightOnHover horizontalSpacing='xl' verticalSpacing='sm'>
        <thead>
          <tr>
            <th>Position</th>
            <th>Club</th>
            <th>Played</th>
            <th>Won</th>
            <th>Drawn</th>
            <th>Lost</th>
            <th>GF</th>
            <th>GA</th>
            <th>GD</th>
            <th>Points</th>
          </tr>
        </thead>
        <tbody>
          {tableData.map((team, index) => (
            <tr key={index}>
              <td>{team.rank}</td>
              <td>
                <Text color='green' component='a' href={`/fixtures/${team.name}`}>
                  {team.name}
                </Text>
              </td>
              <td>{team.played}</td>
              <td>{team.won}</td>
              <td>{team.drawn}</td>
              <td>{team.lost}</td>
              <td>{team.gf}</td>
              <td>{team.ga}</td>
              <td>{team.gd}</td>
              <td>{team.points}</td>
            </tr>
          ))}
        </tbody>
      </Table>
    </Styled>
  )
}

const Styled = styled(Container)`
  padding-top: 100px;
`

export default LeagueTable
