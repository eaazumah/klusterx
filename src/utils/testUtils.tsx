import { configureStore } from '@reduxjs/toolkit'
import '@testing-library/jest-dom'
import { render as rtlRender, renderHook as rtlRenderHook } from '@testing-library/react'
import 'jest-styled-components'
import React from 'react'
import { Provider } from 'react-redux'
import settings from '../redux/settingsSlice'
import { IReduxSate } from '../types/declarations'

interface Options {
  store?: any
  preloadedState?: Partial<IReduxSate>
}

interface OptionsWithRouter {
  store?: any
  preloadedState?: Partial<IReduxSate>
}

const render = (
  ui: React.ReactElement,
  {
    preloadedState,
    store = configureStore({
      preloadedState,
      reducer: { settings },
    }),
    ...options
  }: Options = {},
) => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...options })
}

const renderWithRouter = (
  ui: React.ReactElement,
  {
    preloadedState,
    store = configureStore({
      preloadedState,
      reducer: { settings },
    }),
    ...options
  }: OptionsWithRouter = {},
) => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...options })
}

const renderHook = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  {
    mocks = [],
    preloadedState,
    store = configureStore({
      preloadedState,
      reducer: { settings },
    }),
    ...options
  }: any = {},
) => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRenderHook(callback, {
    ...options,
    wrapper: Wrapper,
  })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render, renderHook, renderWithRouter }
