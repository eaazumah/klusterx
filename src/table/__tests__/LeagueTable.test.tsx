import { render } from '../../utils/testUtils'
import LeagueTable from '../LeagueTable'

describe('LeagueTable', () => {
  test('should render table ', () => {
    const { container } = render(<LeagueTable />)
    expect(container).toMatchSnapshot()
  })
})
