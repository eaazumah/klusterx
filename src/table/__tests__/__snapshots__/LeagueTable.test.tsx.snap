// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeagueTable should render table  1`] = `
.c0 {
  padding-top: 100px;
}

<div>
  <div
    class="mantine-Container-root c0 mantine-1pdibkf"
  >
    <div
      class="mantine-Badge-root mantine-1q104z4"
    >
      <span
        class="mantine-1t45alw mantine-Badge-inner"
      >
        League Table
      </span>
    </div>
    <div
      class="mantine-1cw9ra3"
    />
    <table
      class="mantine-Table-root mantine-1hu3wr5"
      data-hover="true"
      data-striped="true"
    >
      <thead>
        <tr>
          <th>
            Position
          </th>
          <th>
            Club
          </th>
          <th>
            Played
          </th>
          <th>
            Won
          </th>
          <th>
            Drawn
          </th>
          <th>
            Lost
          </th>
          <th>
            GF
          </th>
          <th>
            GA
          </th>
          <th>
            GD
          </th>
          <th>
            Points
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            <a
              class="mantine-Text-root mantine-bwhnbm"
              href="/fixtures/Leicester City"
            >
              Leicester City
            </a>
          </td>
          <td>
            6
          </td>
          <td>
            3
          </td>
          <td>
            3
          </td>
          <td>
            0
          </td>
          <td>
            8
          </td>
          <td>
            3
          </td>
          <td>
            5
          </td>
          <td>
            12
          </td>
        </tr>
        <tr>
          <td>
            2
          </td>
          <td>
            <a
              class="mantine-Text-root mantine-bwhnbm"
              href="/fixtures/Liverpool"
            >
              Liverpool
            </a>
          </td>
          <td>
            6
          </td>
          <td>
            3
          </td>
          <td>
            3
          </td>
          <td>
            0
          </td>
          <td>
            8
          </td>
          <td>
            4
          </td>
          <td>
            4
          </td>
          <td>
            12
          </td>
        </tr>
        <tr>
          <td>
            3
          </td>
          <td>
            <a
              class="mantine-Text-root mantine-bwhnbm"
              href="/fixtures/Manchester United"
            >
              Manchester United
            </a>
          </td>
          <td>
            6
          </td>
          <td>
            1
          </td>
          <td>
            3
          </td>
          <td>
            2
          </td>
          <td>
            11
          </td>
          <td>
            9
          </td>
          <td>
            2
          </td>
          <td>
            6
          </td>
        </tr>
        <tr>
          <td>
            4
          </td>
          <td>
            <a
              class="mantine-Text-root mantine-bwhnbm"
              href="/fixtures/Tottenham Hotspur"
            >
              Tottenham Hotspur
            </a>
          </td>
          <td>
            6
          </td>
          <td>
            0
          </td>
          <td>
            6
          </td>
          <td>
            0
          </td>
          <td>
            3
          </td>
          <td>
            3
          </td>
          <td>
            0
          </td>
          <td>
            6
          </td>
        </tr>
        <tr>
          <td>
            5
          </td>
          <td>
            <a
              class="mantine-Text-root mantine-bwhnbm"
              href="/fixtures/Manchester City"
            >
              Manchester City
            </a>
          </td>
          <td>
            6
          </td>
          <td>
            1
          </td>
          <td>
            3
          </td>
          <td>
            2
          </td>
          <td>
            4
          </td>
          <td>
            5
          </td>
          <td>
            -1
          </td>
          <td>
            6
          </td>
        </tr>
        <tr>
          <td>
            6
          </td>
          <td>
            <a
              class="mantine-Text-root mantine-bwhnbm"
              href="/fixtures/Chelsea"
            >
              Chelsea
            </a>
          </td>
          <td>
            6
          </td>
          <td>
            0
          </td>
          <td>
            4
          </td>
          <td>
            2
          </td>
          <td>
            5
          </td>
          <td>
            10
          </td>
          <td>
            -5
          </td>
          <td>
            4
          </td>
        </tr>
        <tr>
          <td>
            7
          </td>
          <td>
            <a
              class="mantine-Text-root mantine-bwhnbm"
              href="/fixtures/Arsenal"
            >
              Arsenal
            </a>
          </td>
          <td>
            6
          </td>
          <td>
            0
          </td>
          <td>
            4
          </td>
          <td>
            2
          </td>
          <td>
            5
          </td>
          <td>
            10
          </td>
          <td>
            -5
          </td>
          <td>
            4
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
